waspc/src/Wasp/JsImport.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+
+module Wasp.JsImport
+  ( JsImport (..),
+    JsImportName (..),
+    JsImportData,
+    JsImportAlias,
+    JsImportPath,
+    getJsImportData,
+  )
+where
+
+import Data.Data (Data)
+import Data.List (isPrefixOf)
+import StrongPath (Dir', File', Path, Posix, Rel)
+import qualified StrongPath as SP
+
+data JsImport = JsImport
+  { -- | What is being imported.
+    _name :: JsImportName,
+    -- | Path from which we are importing.
+    _path :: JsImportPath,
+    -- | Alias for the imported name
+    _importAlias :: Maybe JsImportAlias
+  }
+  deriving (Show, Eq, Data)
+
+type JsImportPath = Path Posix (Rel Dir') File'
+
+type Identifier = String
+
+type JsImportAlias = String
+
+data JsImportName
+  = -- | Represents external imports like @import Identifier from "file.js"@
+    JsImportModule Identifier
+  | -- | Represents external imports like @import { Identifier } from "file.js"@
+    JsImportField Identifier
+  deriving (Show, Eq, Data)
+
+type JsImportIdentifier = String
+
+type JsImportWhat = String
```

Comment:
I know I gave it this name long time ago but looking at it now I think it is a pretty bad name :D. I am not sure how to better name it though.

Reply:
Looking at JS AST tools, they call this part "importClause". I'll go with that 👍

<img width="503" alt="Screenshot 2023-01-31 at 16 00 10" src="https://user-images.githubusercontent.com/2223680/215795930-6656e569-e41e-411d-a847-c08c394d8b66.png">